pkg/transformers/config/defaultconfig/image.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaultconfig
+
+const (
+	imageFieldSpecs = `
+image:
+- path: containers
+- path: initContainers
```

Comment:
```
- path: containers
- path: initContainers
```
These paths are different from the paths in other transformer configurations, where the path from the root is listed, for example
```
spec/template/metadata/annotations
```

I suggest to keep consistent with other fieldSpecs. Then when users want to customize the image transformer configuration, they provide it similarly to other transformers, such as
```
- path: spec/mycontainer
   kind: MyApplication
```

Reply:
Agree

Reply:
The only concern is that it may break backward compatibility in some way. Since we can not enumerate all possible path for `containers` and `initContainers`, though we can add the most common cases to default configuration. Is that good for you?

Reply:
What about keeping `containers` and `initContainers` hard coded in the  image transformer code and loading extra path for images from the image transformer configurations?

Reply:
Sounds good to me.

/assign yujunz

Reply:
Done @Liujingfang1